paddle/fluid/framework/ir/pass.h

```diff
@@ -135,16 +140,26 @@ class Pass {
     required_pass_attrs_.insert(attrs.begin(), attrs.end());
   }
 
+  void RegisterDefaultPassAttrs(const std::unordered_set<std::string> &attrs) {
+    default_pass_attrs_.insert(attrs.begin(), attrs.end());
+  }
+
   void RegisterRequiredGraphAttrs(
       const std::unordered_set<std::string> &attrs) {
     required_graph_attrs_.insert(attrs.begin(), attrs.end());
   }
 
+  // Pass doesn't take ownership. PassRegistrar should delete default_attrs
+  void CopyDefaultAttrs(std::map<std::string, boost::any> default_attrs_) {
+    attrs_.insert(default_attrs_.begin(), default_attrs_.end());
+  }
+
   void RegisterType(const std::string &type) { type_ = type; }
 
   mutable bool applied_{false};
   std::string type_;
   std::unordered_set<std::string> required_pass_attrs_;
+  std::unordered_set<std::string> default_pass_attrs_;
```

Comment:
IMO the name `default_attrs_` would be sufficient as it is already in the `Pass` class.